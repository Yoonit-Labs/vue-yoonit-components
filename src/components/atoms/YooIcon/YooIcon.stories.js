import YooAvatar from './YooAvatar.vue'
import PropsConfig from '@/config/Props.config'

export default {
  title: 'Yoonit/YooAvatar',
  component: YooAvatar,
  argTypes: {
    variation: {
      control: {
        type: 'select',
        options: PropsConfig.variation.options
      }
    },
    buttonSize: {
      control: {
        type: 'select',
        options: PropsConfig.buttonSize.options
      }
    },
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    iconPosition: {
      control: {
        type: 'select',
        options: PropsConfig.iconPosition.options
      }
    },
    iconSize: {
      control: {
        type: 'select',
        options: PropsConfig.iconSize.options
      }
    },
    iconStyle: {
      control: {
        type: 'select',
        options: PropsConfig.iconStyle.options
      }
    },
    animation: {
      control: {
        type: 'select',
        options: PropsConfig.animation.options
      }
    },
    iconCustomColor: {
      control: {
        type: 'text'
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { YooAvatar },
  template:
    '<yoo-avatar v-bind="$props"></yoo-avatar>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Avatar' }

export const Outline = Template.bind({})
Outline.args = { text: 'Yoo Avatar', variation: 'outline' }
