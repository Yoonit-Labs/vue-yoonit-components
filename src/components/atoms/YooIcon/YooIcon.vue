<template lang="pug">
  i(
    :class="['yoo-icon', takeIconStyle, takeIconName, ...takeModifier]"
  )
</template>

<script>

export default {
  name: 'YooIcon',
  props: {
    icon: {
      type: String,
      default: ''
    },
    iconStyle: {
      type: String,
      default: 'solid',
      validator: value =>
        [
          'solid',
          'regular'
        ]
          .includes(value)
    },
    event: {
      type: String,
      required: false,
      default: ''
    },
    size: {
      type: String,
      default: 'md',
      validator: value =>
        [
          'xxxs',
          'xxs',
          'xs',
          'sm',
          'md',
          'lg',
          'xl',
          'xxl',
          'xxxl',
          'display'
        ]
          .includes(value)
    },
    fill: {
      type: String,
      default: 'neutral',
      validator: value =>
        [
          'primary',
          'danger',
          'neutral',
          'light',
          'dark'
        ]
          .includes(value)
    }
  },
  data: () => ({}),
  created () {},
  mounted () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  components: {},
  computed: {
    takeIconName () {
      return `fa-${this.icon}`
    },
    takeIconStyle () {
      return this.iconStyle === 'solid'
        ? 'fas' // fontawesome solid
        : 'far' // fontawesome regular
    },
    takeModifier () {
      const block = 'yoo-icon'
      const classList = []

      if (this.size) {
        classList
          .push(
            `${block}__size--${this.size}`
          )
      }

      if (this.fill) {
        classList
          .push(
            `${block}__fill--${this.fill}`
          )
      }
      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooIcon.sass" lang="sass" scoped></style>
