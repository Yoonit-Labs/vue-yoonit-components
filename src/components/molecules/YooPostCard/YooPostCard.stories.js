import yooPostCard from './YooPostCard.vue'
import PropsConfig from './YooPostCard.config'

export default {
  title: 'Yoonit/PostCard',
  component: yooPostCard,
  argTypes: {
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    iconSize: {
      control: {
        type: 'select',
        options: PropsConfig.iconSize.options
      }
    },
    iconFill: {
      control: {
        type: 'select',
        options: PropsConfig.iconFill.options
      }
    },
    iconStyle: {
      control: {
        type: 'select',
        options: PropsConfig.iconStyle.options
      }
    },
    appIndicatorFill: {
      control: {
        type: 'select',
        options: PropsConfig.appIndicatorFill.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooPostCard },
  template:
    '<yoo-post-card v-bind="$props"></yoo-post-card>'
})

export const Custom = Template.bind({})
Custom.args = { title: 'Yoo PostCard', subtitle: 'Test' }

export const Danger = Template.bind({})
Danger.args = { title: 'Yoo PostCard', subtitle: 'Test', fill: 'danger', showIndicator: true, icon: 'cog' }

export const Neutral = Template.bind({})
Neutral.args = { title: 'Yoo PostCard', subtitle: 'Test', fill: 'neutral', showIndicator: true, icon: 'cog' }

export const Light = Template.bind({})
Light.args = { title: 'Yoo PostCard', subtitle: 'Test', fill: 'light', showIndicator: true, icon: 'cog' }

export const Dark = Template.bind({})
Dark.args = { title: 'Yoo PostCard', subtitle: 'Test', fill: 'dark', showIndicator: true, icon: 'cog' }
