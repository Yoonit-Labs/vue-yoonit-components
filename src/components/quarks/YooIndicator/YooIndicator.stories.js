import yooIndicator from './YooIndicator.vue'
import PropsConfig from './YooIndicator.config'

export default {
  title: 'Yoonit/Indicator',
  component: yooIndicator,
  argTypes: {
    fill: {
      control: {
        type: 'select',
        options: PropsConfig.fill.options
      }
    },
    size: {
      control: {
        type: 'select',
        options: PropsConfig.size.options
      }
    }
  }
}

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { yooIndicator },
  template:
    '<yoo-indicator v-bind="$props"></yoo-indicator>'
})

export const Custom = Template.bind({})
Custom.args = { text: 'Yoo Indicator test component' }

export const Default = Template.bind({})
Default.args = { text: '1' }

export const Primary = Template.bind({})
Primary.args = { text: '1', fill: 'primary' }

export const Light = Template.bind({})
Light.args = { text: '2', fill: 'light' }

export const Dark = Template.bind({})
Dark.args = { text: '10', fill: 'dark' }

export const Neutral = Template.bind({})
Neutral.args = { text: '5', fill: 'neutral' }
