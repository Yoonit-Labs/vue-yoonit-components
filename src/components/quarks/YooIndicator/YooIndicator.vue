<template lang="pug">
  div(
    :class="['yoo-indicator', ...takeModifier]"
  ) {{ text }}
</template>

<script>

export default {
  name: 'YooIndicator',
  props: {
    fill: {
      type: String,
      required: false,
      default: 'danger',
      validator: value =>
        [
          'primary',
          'danger',
          'light',
          'dark',
          'neutral'
        ]
          .includes(value)
    },
    size: {
      type: String,
      required: false,
      default: 'normal',
      validator: value =>
        [
          'small',
          'medium',
          'normal',
          'large'
        ]
          .includes(value)
    },
    animation: {
      type: Boolean,
      required: false,
      default: true
    },
    text: {
      type: String,
      required: false,
      default: ''
    }
  },
  components: {},
  data: () => ({}),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {},
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {
    takeModifier () {
      const classList = [`yoo-indicator--${this.size}`, `yoo-indicator--${this.fill}`]

      if (this.animation) {
        classList
          .push('yoo-indicator--pulse')
      }

      return classList
    }
  },
  methods: {},
  filters: {},
  watch: {}
}
</script>

<style src="./YooIndicator.sass" lang="sass" scoped></style>
