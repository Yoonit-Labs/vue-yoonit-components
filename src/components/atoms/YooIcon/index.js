import YooIcon from './YooIcon.vue'
export default YooIcon
