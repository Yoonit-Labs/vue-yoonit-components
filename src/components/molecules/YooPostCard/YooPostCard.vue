<template lang="pug">
  div(
    :class="['yoo-post-card', takeModifier]"
    @click="doClick"
  )
    YooFlexLayout(
      row="1"
      col="1"
      width="100%"
      justifyContent="space-between"
    )
      YooIcon(
        :icon="icon"
        :fill="iconFill"
        :size="iconSize"
        :iconStyle="iconStyle"
        class="yoo-post-card__icon"
      )

      YooIndicator(
        v-if="showIndicator"
        size="small"
        :fill="appIndicatorFill"
      )

    h3(
      row="2"
      col="1"
      class="yoo-post-card__title"
    ) {{ title }}

    p(
      row="3"
      col="1"
      class="yoo-post-card__subtitle"
    ) {{ subtitle }}
</template>

<script>

import YooIcon from '@/components/atoms/YooIcon/YooIcon.vue'
import YooFlexLayout from '@/components/quarks/YooFlexLayout/YooFlexLayout.vue'
import YooIndicator from '@/components/quarks/YooIndicator/YooIndicator.vue'

import PropsConfig from '@/components/molecules/YooPostCard/YooPostCard.config'

export default {
  name: 'YooPostCard',
  props: {
    title: {
      type: String,
      default: ''
    },
    subtitle: {
      type: String,
      default: ''
    },
    fill: {
      type: String,
      default: 'white',
      validator: value => PropsConfig.fill.options.includes(value)
    },
    showIndicator: {
      type: Boolean,
      default: false
    },
    icon: {
      type: String,
      default: ''
    },
    iconSize: {
      type: String,
      default: 'md',
      validator: value => PropsConfig.iconSize.options.includes(value)
    },
    iconFill: {
      type: String,
      default: 'neutral',
      validator: value => PropsConfig.iconFill.options.includes(value)
    },
    iconStyle: {
      type: String,
      default: 'solid',
      validator: value => PropsConfig.iconStyle.options.includes(value)
    },
    appIndicatorFill: {
      type: String,
      default: 'danger',
      validator: value => PropsConfig.appIndicatorFill.options.includes(value)
    }
  },
  components: {
    YooFlexLayout,
    YooIndicator,
    YooIcon
  },
  data: () => ({}),
  beforeCreate () {},
  created () {},
  beforeMount () {},
  mounted () {},
  beforeUpdate () {},
  updated () {},
  beforeDestroy () {},
  destroyed () {},
  computed: {
    /**
    * @description Print classes based on the chosen props
    * @computed takeModifier
    * @returns {array}
    */
    takeModifier () {
      return `yoo-post-card--fill-${this.fill}`
    }
  },
  methods: {
    doClick (event) {
      if (event) {
        this.$emit('onClick', event)
      }
    }
  },
  filters: {},
  watch: {}
}
</script>

<style src="./YooPostCard.sass" lang="sass" scoped></style>
