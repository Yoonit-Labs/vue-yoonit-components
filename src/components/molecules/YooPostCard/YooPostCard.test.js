import { shallowMount } from '@vue/test-utils'
import YooPostCard from '@/components/molecules/YooPostCard/YooPostCard.vue'
import YooIndicator from '@/components/quarks/YooIndicator/YooIndicator.vue'
import YooIcon from '@/components/atoms/YooIcon/YooIcon.vue'
import PropsConfig from './YooPostCard.config'

const classBlock = 'yoo-post-card'
const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(YooPostCard, { components: { YooIndicator } }, {
    slots: { default: SlotText }
  })
}

describe('YooPostCard Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', async () => {
    expect(wrapper.classes('yoo-post-card')).toBe(true)
    expect(wrapper.find('.yoo-post-card__icon').exists()).toBe(true)
    // Test check YooIndicator
    wrapper = shallowMount(YooIndicator)
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.yoo-indicator').exists()).toBe(true)
  })

  describe('Props', () => {
    describe('fill', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.fill.options.includes(YooPostCard.props.fill.default)).toBe(true)
      })
      PropsConfig.fill.options.forEach(fill => {
        it(`Includes fill class: .${classBlock}--fill-${fill}`, async () => {
          await wrapper.setProps({ fill })
          expect(wrapper.find(`.${classBlock}--fill-${fill}`).exists()).toBe(true)
        })
      })
    })
  }) // describe Props

  describe('Events', () => {
    describe('doClick', () => {
      it('Emits doClick Event', async () => {
        wrapper.find(`.${classBlock}`).trigger('click')
        expect(wrapper.emitted()).toHaveProperty('onClick')
      })
    })
  }) // describe Events
})
