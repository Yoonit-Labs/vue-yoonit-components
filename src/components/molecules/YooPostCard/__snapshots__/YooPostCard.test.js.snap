// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooPostCard Component Matches Snapshot 1`] = `
<div class="yoo-post-card yoo-post-card--fill-white">
  <div class="yoo-flex-layout" style="flex-direction: row; flex-wrap: nowrap; justify-content: space-between; align-items: stretch; align-content: stretch;" row="1" col="1" width="100%"><i class="yoo-post-card__icon yoo-icon fas fa- yoo-icon__size--md yoo-icon__fill--neutral"></i>
    <div class="yoo-indicator yoo-indicator--small yoo-indicator--danger yoo-indicator--pulse"></div>
  </div>
  <h3 row="2" col="1" class="yoo-post-card__title"></h3>
  <p row="3" col="1" class="yoo-post-card__subtitle"></p>
</div>
`;
